tests/test-buttercup.el

```diff
@@ -501,17 +501,40 @@
 
 (describe "The Spy "
   (let (test-function)
-    (before-each
+    (before-all
```

Comment:
It's explained in the commit message: https://github.com/jorgenschaefer/emacs-buttercup/pull/95/commits/653f7edc2f431a0ebb983a8d3624ad7c24b9adae

Basically, some of the tests aren't actually testing anything if you use `before-each`. For example, the "restores the old value after a spec run" will always pass because the `before-each` will restore the old value even if `spy-on` doesn't.